provider/core/src/any.rs

```diff
@@ -350,12 +350,31 @@ pub trait AnyProvider {
     fn load_any(&self, key: DataKey, req: DataRequest) -> Result<AnyResponse, DataError>;
 }
 
+impl<'a, T: AnyProvider + ?Sized> AnyProvider for &'a T {
+    fn load_any(&self, key: DataKey, req: DataRequest) -> Result<AnyResponse, DataError> {
+        (**self).load_any(key, req)
+    }
+}
+
 impl<T: AnyProvider + ?Sized> AnyProvider for alloc::boxed::Box<T> {
     fn load_any(&self, key: DataKey, req: DataRequest) -> Result<AnyResponse, DataError> {
         (**self).load_any(key, req)
     }
 }
 
+impl<T: AnyProvider + ?Sized> AnyProvider for alloc::rc::Rc<T> {
+    fn load_any(&self, key: DataKey, req: DataRequest) -> Result<AnyResponse, DataError> {
+        (**self).load_any(key, req)
+    }
+}
+
+#[cfg(target_has_atomic = "ptr")]
```

Comment:
thought: hm, in general we haven't been careful about this, but I guess it doesn't hurt to start?

Reply:
I think in other places we've been using `#[cfg(feature = "std")]` for gating `Arc`. I'm not sure which of these is better though. I think the main thing is to not mess up the compilation of targets like `wasm32-unknown-unknown`. If this works without needing the `std` feature then that's great and we should use this pattern in more places.